sklearn/cluster/bicluster.py

```diff
@@ -156,6 +169,8 @@ def _svd(self, array, n_components, n_discard):
 
         assert_all_finite(u)
         assert_all_finite(vt)
+        if u.shape[1] == 1 and vt.shape[0] == 1:
```

Comment:
When there is only 1 singular vector, then it is also getting discarded and the shape of u becomes ``[n_samples, 0]`` and v becomes ``[0, n_features]`` and later operations are failing the tests.